src/main/java/org/subethamail/smtp/internal/proxy/ProxyProtocolV2Handler.java

```diff
@@ -0,0 +1,352 @@
+package org.subethamail.smtp.internal.proxy;
+
+import static org.subethamail.smtp.internal.util.HexUtils.toHex;
+import com.github.davidmoten.guavamini.Preconditions;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.subethamail.smtp.internal.util.ArrayUtils;
+import org.subethamail.smtp.server.Session;
+
+/**
+ * Implements {@link ProxyHandler} for <a href="https://www.haproxy.org/download/2.3/doc/proxy-protocol.txt">PROXY
+ * protocol V2 binary</a>.
+ *
+ * @author Diego Salvi
+ */
+public class ProxyProtocolV2Handler implements ProxyHandler {
+    
+    private final static Logger LOG = LoggerFactory.getLogger(ProxyProtocolV2Handler.class);
+    
+    /**
+     * Default maximum data length. Standard max data size in 216 for unix socket (two unix address, 108 bytes each).
+     * 2048 is reasonable default to host data plus some optional informations.
+     */
+    public static final int DEFAULT_MAX_DATA_LENGTH = 2048;
```

Comment:
make private and add no-arg constructor with javadoc about the default value. More discoverable like that?